src/main/java/com/nimbusds/jose/crypto/ECDH1PUX25519Encrypter.java

```diff
@@ -0,0 +1,216 @@
+ *
+ * <ul>
+ *     <li>{@link com.nimbusds.jose.EncryptionMethod#A128CBC_HS256}
+ *     <li>{@link com.nimbusds.jose.EncryptionMethod#A192CBC_HS384}
+ *     <li>{@link com.nimbusds.jose.EncryptionMethod#A256CBC_HS512}
+ * </ul>
+ *
+ * @author Alexander Martynov
+ * @version 2021-08-03
+ */
+@ThreadSafe
+public class ECDH1PUX25519Encrypter extends ECDH1PUCryptoProvider implements JWEEncrypter {
+
+
+	/**
+	 * The public key.
+	 */
+	private final OctetKeyPair publicKey;
+
+	/**
+	 * The private key.
+	 */
+	private final OctetKeyPair privateKey;
+
+
+	/**
+	 * Creates a new Curve25519 Elliptic Curve Diffie-Hellman encrypter.
+	 *
+	 * @param privateKey The private key. Must not be {@code null}.
+	 * @param publicKey The public key. Must not be {@code null}.
+	 *
+	 * @throws JOSEException If the key subtype is not supported.
+	 */
+	public ECDH1PUX25519Encrypter(final OctetKeyPair privateKey, final OctetKeyPair publicKey)
+			throws JOSEException {
+
+		super(publicKey.getCurve());
+
+		if (! Curve.X25519.equals(privateKey.getCurve())) {
+			throw new JOSEException("X25519Encrypter only supports OctetKeyPairs with crv=X25519");
+		}
+
+		if (! Curve.X25519.equals(publicKey.getCurve())) {
+			throw new JOSEException("X25519Encrypter only supports OctetKeyPairs with crv=X25519");
+		}
+
+		if (publicKey.isPrivate()) {
+			throw new JOSEException("X25519Encrypter requires a public key, use OctetKeyPair.toPublicJWK()");
+		}
+
+		this.publicKey = publicKey;
+		this.privateKey = privateKey;
+	}
+
+
+	@Override
+	public Set<Curve> supportedEllipticCurves() {
+
+		return Collections.singleton(Curve.X25519);
+	}
+
+
+	/**
+	 * Returns the public key.
+	 *
+	 * @return The public key.
+	 */
+	public OctetKeyPair getPublicKey() {
+
+		return publicKey;
+	}
+
+	/**
+	 * Returns the private key.
+	 *
+	 * @return The private key.
+	 */
+	public OctetKeyPair getPrivateKey() {
+
+		return privateKey;
+	}
+
+	@Override
+	public JWECryptoParts encrypt(final JWEHeader header, final byte[] clearText)
+			throws JOSEException {
+
+		// Generate ephemeral X25519 key pair
+		final byte[] ephemeralPrivateKeyBytes = X25519.generatePrivateKey();
+		final byte[] ephemeralPublicKeyBytes;
+		try {
+			ephemeralPublicKeyBytes = X25519.publicFromPrivate(ephemeralPrivateKeyBytes);
+
+		} catch (InvalidKeyException e) {
+			// Should never happen since we just generated this private key
+			throw new JOSEException(e.getMessage(), e);
+		}
+
+		final OctetKeyPair ephemeralPrivateKey =
+				new OctetKeyPair.Builder(getCurve(), Base64URL.encode(ephemeralPublicKeyBytes)).
+						d(Base64URL.encode(ephemeralPrivateKeyBytes)).
+						build();
+		final OctetKeyPair ephemeralPublicKey = ephemeralPrivateKey.toPublicJWK();
+
+		// Add the ephemeral public EC key to the header
+		JWEHeader updatedHeader = new JWEHeader.Builder(header).
+				ephemeralPublicKey(ephemeralPublicKey).
+				build();
+
+		// Derive 'Z'
```

Comment:
Comment doesn't seem aligned with the actual derivation, and I'm dubious it adds any value beyond the method names themselves!